### PR TITLE
Releasing new stable API version migrating ARE from preview to feature 

### PR DESCRIPTION
**Pull request by sushil490023**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ X] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [X ] Other, please clarify:
    - We are moving an preview API feature to Stable release

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in #23988

**Comment by raosuhas** [comment]
Can you please update the description checkboxes ? 

For this PR you need to follow the instructions there https://eng.ms/docs/products/azure-developer-experience/design/api-specs/api-specs?tabs=openapi to add a base commit with the previous api version.

**Comment by sushil490023** [comment]
Hi @raosuhas 

 Thanks for reviewing

I have updated the PR description checkboxes.

In the first commit, I simply copied the preview API Swagger version 2023-05-15-preview to a stable API version 2024-10-23. The file in this version contains no changes except for the updated API version. Subsequent commits include the required changes.
 
I also have a few questions regarding the test failures in the PR:

Avocado Errors
The Avocado tests are failing for the default API version. However, the new API versions do not include dscCompilationJob.json and powershell72Module.json APIs.


Swagger LintDiff Failure
Most validation errors in Swagger LintDiff are related to older Swagger versions, not the new one.

Additionally, could you please:
1. Add the IDC tag to this PR?
2. Grant permission to run the tests by commenting /azp run.
Thanks!

**Comment by ms-bogdan-rotaru** [comment]
@sushil490023 Can you please provide an explanation for the errors reported by the Swagger LintDiff check and how were these resolved in the preview API Swagger version?

**Comment by sushil490023** [comment]
@ms-bogdan-rotaru These errors were previously not coming for when it was merged during preview API version

**Comment by venu-sivanadham** [comment]
@sushil490023 please address the LintDiff error or if you have valid reasons to continue with these errors then take a look at the suppression guide [here](https://aka.ms/azsdk/pr-suppressions)

**Comment by sushil490023** [comment]
Hi @venu-sivanadham ,

For Lint diff erros
1. ProvisioningStateValidation errors are flagged for **ProvisioningState must have terminal states: Succeeded, Failed and Canceled.**  its coming as previous api version are having **Cancelled instead of Canceled** , this cannot be changed as been already being used.
2. [AvoidAdditionalProperties](https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-additional-properties.md) errors are flagged with error **Definitions must not have properties named additionalProperties except for user defined tags or predefined references.** . This is being used in production currently  and being consumed

**Comment by viveksheelsingh** [comment]
> > ```
> > reason: This error format is inherited from previous api, cannot change it as it will result in inconsistency.
> > ```
> 
> I have couple of observations that's not aligning with this justification.
> 
> * The package.json swagger file in this same api-version has 'Canceled' enum.
> * In previous stable version (2023-11-01) the enum is defined as 'Canceled' but you are changing it to 'Cancelled' for this stable version.
> 
> Also, you can join ARM office hours for the suppression review.
> 
> Refers to: specification/automation/resource-manager/readme.md:678 in [82b0b81](https://github.com/Azure/azure-rest-api-specs/commit/82b0b810eaeefca73a528e6c8395030710deeba1). [](commit_id = [82b0b81](https://github.com/Azure/azure-rest-api-specs/commit/82b0b810eaeefca73a528e6c8395030710deeba1), deletion_comment = False)

Hi @sushil490023  -- Looks like this is pending a supression review ? Pl attend the IDC office hours for the same: https://outlook.office365.com/book/ARMOfficeHours1@microsoft.onmicrosoft.com/s/DbvLwCX1IUqeyGw6ziXDOQ2?login_hint

File: specification/automation/resource-manager/readme.md [hunk 1]
```diff
@@ -668,7 +668,7 @@ directive:
     from: powershell72Module.json
   - suppress: AvoidAdditionalProperties
     from: runtimeEnvironment.json
-    where: $.definitions.DefaultPackages.additionalProperties
+    where: $.definitions.DefaultPackages.
```
Comment (ajaygit158) [inline specification/automation/resource-manager/readme.md:671]: <nit> You can remove the additional dot at end.

**Review by venu-sivanadham — APPROVED** [review]
:shipit:

**Pull request merged**